Tools/cases_generator/parsing.py

```diff
@@ -458,10 +475,27 @@ def cache_effect(self) -> CacheEffect | None:
                     return CacheEffect(tkn.text, size)
         return None
 
+    def stack_attributes(self) -> list[StackAttribute]:
+        # IDENTIFIER '(' expression ')'
+        res = []
+        while tkn := self.expect(lx.IDENTIFIER):
+            if self.expect(lx.LPAREN):
+                if tkn.text not in STACK_ATTRIBUTES:
+                    raise self.make_syntax_error(f"Stack attribute {tkn.text} is not recognized.")
+                expr = self.expression()
+                assert expr is not None
+                self.require(lx.RPAREN)
+                res.append(StackAttribute(tkn.text.strip(), expr.size.strip()))
+            else:
+                self.backup()
+                break
+        return res
+
     @contextual
     def stack_effect(self) -> StackEffect | None:
-        # IDENTIFIER [':' IDENTIFIER [TIMES]] ['if' '(' expression ')']
-        # | IDENTIFIER '[' expression ']'
+        # stack_attributes IDENTIFIER [':' IDENTIFIER [TIMES]] ['if' '(' expression ')']
+        # | stack_attributes IDENTIFIER '[' expression ']'
+        stack_attributes = self.stack_attributes()
```

Comment:
This will also parse the type attributes for outputs, should that be supported? Do we want error out/warn in that case or is it fine like this?